frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1146,13 +1146,15 @@ function ReaderHighlight:getHighlightBookmarkItem()
         local datetime = os.date("%Y-%m-%d %H:%M:%S")
         local page = self.ui.document.info.has_pages and
                 self.hold_pos.page or self.selected_text.pos0
+        local chapter_name = self.ui.toc:getTocTitleByPage(page)
```

Comment:
OK, as I mentionned in https://github.com/koreader/koreader/issues/6138#issuecomment-626563596, this might be innacurate for CRE documents.
You should try to get the TOC entry whose xpointer is the nearest before your highlight/bookmark xpointer, see https://github.com/koreader/koreader/issues/6138#issuecomment-626543969.
May be by adding a new method to readertoc.lua.

`ReaderToc:getTocTitleByPage(pn_or_xp)` looks like it might be it, but it does like you do, which might be good enough for bookmarked page (although I guess it's already innacurate).
https://github.com/koreader/koreader/blob/12675be0659ad3c1d099610490a4322e3c994fa2/frontend/apps/reader/modules/readerbookmark.lua#L539-L548